scripts/hal_alignment/maf_to_fasta.py

```diff
@@ -0,0 +1,240 @@
+    Unlike ``dir_revhash``, this function only accepts a non-negative integer.
+
+    Given an input non-negative integer (e.g. 1234), this is converted to a
+    string (e.g. '1234'), the first digit is dropped (e.g. '234'), the string
+    is reversed (e.g. '432'), and a Path object is returned in which each of
+    the remaining digits is a path component (e.g. 'Path(4/3/2)').
+
+    Appending this generated path to a base directory enables the creation of
+    a directory where files associated with the specified integer can be created
+    and accessed. Note that because the first digit is dropped, the created
+    directory may contain files corresponding to up to 10 integers, so files
+    associated with a specific integer should be named in a way that allows them
+    to be distinguished from files relevant to other integers (e.g. '1234.tsv').
+
+    Args:
+        non_negative_integer: A non-negative integer.
+
+    Returns:
+        The composed pathlib.Path object.
+
+    Raises:
+        TypeError: If the argument ``non_negative_integer`` is not of integer type,
+            or cannot be converted to an integer string.
+        ValueError: If the argument ``non_negative_integer`` is a negative integer.
+
+    """
+    try:
+        integer_string = format(non_negative_integer, 'd')
+    except (TypeError, ValueError) as exc:
+        patt = re.compile("^(unsupported format string passed to [^.]+[.]__format__"
+                          "|Unknown format code 'd' for object of type '[^']+')$")
+
+        if patt.match(str(exc)):
+            arg_type_name = type(non_negative_integer).__name__
+            msg = f"cannot compose integer-derived path for object of type '{arg_type_name}'"
+            raise TypeError(msg) from exc
+
+        raise exc
+
+    if non_negative_integer < 0:
+        raise ValueError(
+            f'cannot compose integer-derived path for negative integer: {non_negative_integer}')
+
+    return Path(*reversed(integer_string[1:]))
+
+
+if __name__ == '__main__':
+
+    parser = ArgumentParser(description='Convert each block of a MAF alignment to a FASTA file.')
+    parser.add_argument('maf_file', metavar='PATH',
+                        help="Input MAF file with alignment blocks. The src fields of"
+                             "this MAF file should be of the form '<genome>.<seqid>'.")
+    parser.add_argument('output_dir', metavar='PATH',
+                        help='Output directory under which FASTA files will be created.')
+    parser.add_argument('--genomes-file', metavar='PATH',
+                        help="File listing the genomes in the input MAF file, one per line. This is used"
+                             " to compile a regex that splits MAF src fields of the form '<genome>.<seqid>'"
+                             " into their component parts. If any of the genomes or their sequences in the"
+                             " MAF file contains a dot ('.'), this is required.")
```

Comment:
Because we are talking about MAF and FASTA formats, I find confusing using the term sequence in the sentence: `If any of the genomes or their sequences`. For a minute I thought you meant the DNA sequence, and I was a little bit confused. However, I cannot think of a better phrasing/word as of yet.

Reply:
It's possible we have the same understanding of what is meant here by genomes (e.g. `GRCh38`) and sequences (e.g. `chr1`). But perhaps it could be less ambiguous. Maybe something like `If any of the genomes or their DNA assembly sequences`?

Reply:
That sentence is great :+1: